crates/consensus/src/header.rs

```diff
@@ -397,8 +397,8 @@ impl Decodable for Header {
             logs_bloom: Decodable::decode(buf)?,
             difficulty: Decodable::decode(buf)?,
             number: U256::decode(buf)?.to::<u64>(),
-            gas_limit: U256::decode(buf)?.to::<u64>(),
-            gas_used: U256::decode(buf)?.to::<u64>(),
+            gas_limit: U256::decode(buf)?.to::<u128>(),
+            gas_used: U256::decode(buf)?.to::<u128>(),
```

Comment:
I think we should either use u128::decode or use try_into because to:: is not safe